mne_bids/write.py

```diff
@@ -879,6 +883,14 @@ def write_raw_bids(raw, bids_basename, output_path, events_data=None,
     acquisition, task, run = params['acq'], params['task'], params['run']
     kind = _handle_kind(raw)
 
+    if anonymize:
+        if ext != '.fif':
+            warn('Converting to FIF to anonymize')
+            ext = '.fif'
+        raw.info = anonymize_info(raw.info)
+        raw_fname = 'anonymize_tmp-raw.fif'
+        raw.save(raw_fname)
```

Comment:
what if I pass EEG only data?

Reply:
I think then the pending PR will convert but @jasmainak was saying anonymize_info requires a FIF file and since it is being copied from disc it has to be saved out. I think in practice what this means is if you want to anonymize a BV EEG file it goes to FIF and back but I'm not sure there's a better way without writing anonymizers for each file type...